packages/language/src/parser/pli-lexer.ts

```diff
@@ -0,0 +1,58 @@
+/**
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ *
+ */
+
+import { DefaultLexer, LexerResult } from "langium";
+
+const NEWLINE = '\n'.charCodeAt(0);
+
+export class Pl1Lexer extends DefaultLexer {
+
+    override tokenize(text: string): LexerResult {
+        const lines = this.splitLines(text);
+        const adjustedLines = lines.map(line => this.adjustLine(line));
+        const adjustedText = adjustedLines.join('');
+        return super.tokenize(adjustedText);
+    }
+
+    private splitLines(text: string): string[] {
+        const lines: string[] = [];
+        for (let i = 0; i < text.length; i++) {
+            const start = i;
+            while (i < text.length && text.charCodeAt(i) !== NEWLINE) {
+                i++;
+            }
+            lines.push(text.substring(start, i + 1));
+        }
+        return lines;
+    }
+
+    private adjustLine(line: string): string {
+        let eol = '';
+        if (line.endsWith('\r\n')) {
```

Comment:
Can't we just use Node.js `os.EOL` constant?

Reply:
No, because `os.EOL` is OS dependant. Meanwhile, a Linux user could have a Windows encoded file on their system and the reverse could also be true. Using `os.EOL` would disregard that edge case and lead to incorrect offsets in the lexer.